backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxSparkPlanExecApi.scala

```diff
@@ -553,30 +553,19 @@ class VeloxSparkPlanExecApi extends SparkPlanExecApi {
       partitioning: Partitioning,
       output: Seq[Attribute]): ShuffleWriterType = {
     val conf = GlutenConfig.get
-    // todo: remove isUseCelebornShuffleManager here
-    if (conf.isUseCelebornShuffleManager) {
-      if (conf.celebornShuffleWriterType == ReservedKeys.GLUTEN_SORT_SHUFFLE_WRITER) {
-        if (conf.useCelebornRssSort) {
-          RssSortShuffleWriterType
-        } else if (partitioning != SinglePartition) {
+    SparkEnv.get.shuffleManager match {
+      case shuffleManager: NeedCustomShuffleWriterType =>
```

Comment:
There's a trait `SupportsColumnarShuffle`. We can add and implement method `getColumnarShuffleWriterType` for the supported shuffle managers

Reply:
@marin-ma It's a good idea. But since `ColumnarShuffleManager` is in common module, we must hardcode `velox`/`ch` backendName to distinguish different logic like in this poc code  https://github.com/zjuwangg/gluten/commit/5dc4da65c6f5eeb3057d1377e283d2bba17b74e9. 
It go against our design to decouple backend-velox and common module dependency.
Please ping me if there is better way to refactor this or am I understand wrong

Reply:
> In my opinion, the implementation of sort shuffle writer type for velox+celeborn is a special case. Perhaps we can add a function like `getShuffleWriterTypeForCeleborn` to the BackendsApi to handle this special case, and call it in the `getColumnarShuffleWriterType` in `CelebornShuffleManager`.

Do you mean in this way https://github.com/apache/incubator-gluten/pull/10616/commits/47f2edc4789a3c19e014b48f30984bca986a2ce7 ?

I still think `getColumnarShuffleWriterType` can not put in the `SupportsColumnarShuffle ` but need a separate trait to mark shuffle manage need to do special process